src/Unleash/Utilities/ToggleBootstrapUrlProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Unleash.Internal;
+
+namespace Unleash.Utilities
+{
+    public class ToggleBootstrapUrlProvider : IToggleBootstrapProvider
+    {
+        private readonly CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+        private readonly HttpClient client;
+        private readonly string path;
+        private readonly bool throwOnFail;
+
+        public ToggleBootstrapUrlProvider(string path, bool throwOnFail = false)
+        {
+            this.path = path;
+            client = new HttpClient();
```

Comment:
The HttpClient in dotnet can (and probably should) be reused, it's thread safe but it isn't safe to instantiate over and over again. We've had at least one issue in the past with socket exhaustion due to this (https://github.com/Unleash/unleash-client-dotnet/pull/41/files). The way the client is used here is probably safe. It's almost certainly safe. But I would sleep better at night if this were reused/cached/static

Reply:
Yeah, there's an overload for providing an HttpClient already, specificly to allow reuse (but also to allow configuring custom headers and makes testing easier), but i could definately remove this particular constructor and require a client as parameter. I could also looking at how easy it would be to just reuse the client used in the communication with the Unleash API

Reply:
If you wouldn't mind looking that would be awesome. If it's a huge amount of work then it's probably not worth it and I'll come back to this when it's done (and the other SDKs aren't keeping me busy!) and swap it out later. It's not urgent, like I said, this requires a consumer to do silly things before it goes wrong

Reply:
Yup, i'll have a look and see! Thanks!